### PR TITLE
WIP: Bare metal part 1

### PR DESCRIPTION
**Pull request by justinsb**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2614)

### CODE REVIEW

**Comment by WillemMali** [comment]
You're missing the license headers in some files, you can use `make ci` to be reasonably sure your code will pass the CI tests and you can use `make hooks` to install the CI git pre-commit hook (which you can bypass with `git commit -n`).

File: examples/baremetal/cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+kind: Cluster
+apiVersion: kops/v1alpha1
+metadata:
+  name: baremetal.awsdata.com
+spec:
+  channel: stable
+  cloudProvider: baremetal
+  configBase: s3://clusters.awsdata.com/baremetal.awsdata.com
```
Comment (cdrage) [inline examples/baremetal/cluster.yaml]: Bit confused on the examples, it says aws here yet you're testing baremetal?

**Comment by carldanley** [comment]
Any update on this?

**Comment by justinsb** [comment]
Started pulling out mergeable / reviewable PRs: https://github.com/kubernetes/kops/pull/3291

**Comment by KlavsKlavsen** [comment]
Whats the status on this getting done in kops ? @justinsb  ?

**Comment by olemarkus** [comment]
As mentioned in https://github.com/kubernetes/enhancements/issues/360, this project is no longer active.

**Pull request closed without merging**